src/components/FilterBar/FilterDisplay/Inputs/RelativeDateInput.react.js

```diff
@@ -40,6 +40,14 @@ export class RelativeDateInput extends React.Component {
     this.setState({value: newValue});
   }
 
+  relativeValueSelected(selection) {
+    if(selection === undefined) {
+      selection = this.state.value.value;
```

Comment:
will `this.state.value.value;` be... undefined?

just reading this... if `selection === undefined`, and then we're setting `selection` so would `selection !== undefined` won't ever be `undefined`?

Reply:
This is for when it's called from, where nothing is passed in: https://github.com/jobready/react-filterbar/pull/80/commits/c41a37e48599021a7452361cdc4b753320a2d734#diff-373a351efbbe84e78fefb5f2c835c363R78

I needed to account for both that case and https://github.com/jobready/react-filterbar/pull/80/commits/c41a37e48599021a7452361cdc4b753320a2d734#diff-373a351efbbe84e78fefb5f2c835c363R12, so if no selection is passed in, the value is undefined, at which point we fall back to the value in the current state, which could also be undefined.

Reply:
I found that when setting an empty selection then setting one of the dates, we ended up hitting this method with both an undefined selection and undefined state value.

It comes from these lines here:

```
this.state = { value: this.props.value || { from: null, to: null, value: null } };
this.setDisplayDates(this.props.value['value']);
```

So we have a value (e.g. `{from: 13/08/2019}`), meaning `state.value.value` is undefined, and `this.props.value['value']` is also undefined.
We could possibly do something like `this.setDisplayDates(this.props.value['value'] || null);` so we only need to check for null, but I thought it would be okay to capture those possibilities in `relativeValueSelected`.